### ISSUE TITLE
Add documentation for authentication additions

### ISSUE THREAD

**Issue by ionut-arm** [parallaxsecond/parsec-book#58]
This issue covers bringing the documentation up for the new features around authentication - the `ListAuthenticators` operation and the new peer credentials authenticator. The latter is already handled in #57 - another PR could fix any gaps that were left.

Before starting the work for implementing `ListAuthenticators` we should have at least a draft documentation for how we intend it to work and what its purpose is for a client.

**Comment by hug-dev** [comment]
Is that done in #60?
